pkg/authenticator/ecrsecret_test.go

```diff
@@ -0,0 +1,222 @@
+package authenticator
+
+import (
+	"context"
+	"encoding/base64"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/rest"
+)
+
+func TestAuthFilename(t *testing.T) {
+	config := rest.Config{}
+	t.Run("golden path for set HELM_REGISTRY_CONFIG", func(t *testing.T) {
+		testfile := "/test.txt"
+		os.Setenv("HELM_REGISTRY_CONFIG", testfile)
```

Comment:
Please use the [testing version](https://cs.opensource.google/go/go/+/go1.19:src/testing/testing.go;l=1287), so it's cleaned up afterwards.